src/main/java/com/puppycrawl/tools/checkstyle/utils/JavadocUtils.java

```diff
@@ -74,8 +76,37 @@
     /** Tab pattern. */
     private static final Pattern TAB = Pattern.compile("\t");
 
+    /**
+     * Maps all the supported nested html open tags to their parents in a nested javadoc tree.
+     * For ex: {@code <p> This is a para</p>}. The javadoc tree for such input will be:
+     * {@code
+     * JAVADOC -> JAVADOC [0:0]
+     * |--HTML_ELEMENT -> HTML_ELEMENT [0:0]
+     * |   `--PARAGRAPH -> PARAGRAPH [0:0]
+     * |       |--P_TAG_OPEN -> P_TAG_OPEN [0:0]
+     * |       |   |--OPEN -> < [0:0]
+     * |       |   |--P_HTML_TAG_NAME -> p [0:1]
+     * |       |   `--CLOSE -> > [0:2]
+     * |       |--TEXT ->  This is a para [0:3]
+     * |       `--P_TAG_CLOSE -> P_TAG_CLOSE [0:18]
+     * |           |--OPEN -> < [0:18]
+     * |           |--SLASH -> / [0:19]
+     * |           |--P_HTML_TAG_NAME -> p [0:20]
+     * |           `--CLOSE -> > [0:21]
+     * `--EOF -> <EOF> [0:22]}
+     * As can be seen {@code <p>} is having {@code PARAGRAPH} as parent in the javadoc tree.
+     * This is what is being referred to as nesting. So, {@code <p>} which is
+     * {@link JavadocTokenTypes#P_TAG_OPEN} will be mapped to {@link JavadocTokenTypes#PARAGRAPH}.
+     * Closing all html tags as expected produces a completely nested javadoc tree.
+     * This field isn't lazy initialized because doing so would require a holder class for thread
+     * safety which seems like an unnecessary optimization given the concise memory footprint
+     * and light initialization time of this field.
+     */
+    private static final Map<Integer, Integer> NESTED_HTML_CHILD_PARENT_MAPPINGS = new HashMap<>();
+
     // Using reflection gets all token names and values from JavadocTokenTypes class
     // and saves to TOKEN_NAME_TO_VALUE and TOKEN_VALUE_TO_NAME collections.
+    // -@cs[ExecutableStatementCount] overshoots while populating NESTED_HTML_CHILD_PARENT_MAPPINGS
```

Comment:
Another option would be to remove reference at [JavadocUtils.java#L113](https://github.com/ps-sp/checkstyle/blob/c7a5a65acbb801efcc3f528477a9acf162a791c2/src/main/java/com/puppycrawl/tools/checkstyle/utils/JavadocUtils.java#L113) and directly call `JavadocTokenTypes.class.getDeclaredFields()` in `for` loop at [JavadocUtils.java#L117](https://github.com/ps-sp/checkstyle/blob/c7a5a65acbb801efcc3f528477a9acf162a791c2/src/main/java/com/puppycrawl/tools/checkstyle/utils/JavadocUtils.java#L117)